src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/ISyntaxFactsExtensions.cs

```diff
@@ -248,6 +248,31 @@ public static bool SpansPreprocessorDirective(this ISyntaxFacts syntaxFacts, IEn
         private static bool SpansPreprocessorDirective(this ISyntaxFacts syntaxFacts, SyntaxTriviaList list)
             => list.Any(syntaxFacts.IsPreprocessorDirective);
 
+        public static bool SpansIfOrElseIfPreprocessorDirective(this ISyntaxFacts syntaxFacts, SyntaxToken token)
+        {
+            var parent = token.Parent;
+            return syntaxFacts.SpansIfOrElseIfPreprocessorDirective(parent);
+        }
+        public static bool SpansIfOrElseIfPreprocessorDirective(this ISyntaxFacts syntaxFacts, SyntaxNode? node)
+        {
+            var kinds = syntaxFacts.SyntaxKinds;
+
+            var ifDirectiveKind = kinds.IfDirectiveTrivia;
+            var elseIfDirectiveKind = kinds.ElseIfDirectiveTrivia;
+
+            while (node is not null)
+            {
+                if (node.RawKind == ifDirectiveKind)
+                    return true;
+                if (node.RawKind == elseIfDirectiveKind)
+                    return true;
+
+                node = node.Parent;
+            }
+
+            return false;
+        }
```

Comment:
i don't see the need for these.  aren't you just checking if you're an any directive?

Reply:
These are specifically written for the `#if` and `#elif` directives, which are definite candidates for containing complicated expressions that require traversal of the ancestor nodes.

Reply:
right.  but as i said, i don't know why you just don't look upwards to see if you're ain a directive.  i don't get why we need to specialize any logic as to the tyep of directive you are in.

Reply:
Because `#region`, `#endregion` and others have nothing to do with preprocessing symbols, and this avoids considering text inside such directives be considered a preprocessing symbol identifier.

Reply:
> Because #region, #endregion and others have nothing to do with preprocessing symbols

I don't understand how those are relevant.  How would a match even be found within those?  

Can you show a test that fails with those pp types that demonstrates why this logic is needed (tbh, this is critical for everything in this PR).  Tests should be here that show that if we went hte simple approach that they would now start failing.  Because i'm just not seeing how this could be relevant.  Thanks!